### PR TITLE
Add checkstyle RedundantModifier check and fix offending classes

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @dhalperi

File: examples/java/src/main/java/org/apache/beam/examples/WordCount.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ public String apply(KV<String, Long> input) {
     /**
      * Returns "gs://${YOUR_TEMP_DIRECTORY}/counts.txt" as the default destination.
      */
-    public static class OutputFactory implements DefaultValueFactory<String> {
```
Comment (dhalperi) [inline examples/java/src/main/java/org/apache/beam/examples/WordCount.java]: Note for naive reviewers: "Classes inside of interfaces by definition are public and static, so the public and static modifiers on the inner classes are redundant. On the other hand, classes inside of interfaces can be abstract or non abstract. So, abstract modifier is allowed.:

**Comment by dhalperi** [comment]
Thank you! It actually fixed essentially only redundant interfaces (and their subclasses) which is why I got very confused in review :).

File: sdks/java/build-tools/src/main/resources/beam/checkstyle.xml [hunk 2]
```diff
@@ -111,6 +111,10 @@ page at http://checkstyle.sourceforge.net/config.html -->
       <property name="illegalPkgs" value="avro.shaded, com.google.api.client.repackaged, com.google.appengine.repackaged, io.netty.util.internal"/>
     </module>
 
+    <module name="RedundantModifier">
+      <property name="tokens" value="VARIABLE_DEF, ANNOTATION_FIELD_DEF, CLASS_DEF, ENUM_DEF"/>
```
Comment (swegner) [inline sdks/java/build-tools/src/main/resources/beam/checkstyle.xml]: I'm finding that validating `METHOD_DEF` is also goodness. Will add in another commit shortly..

**Comment by dhalperi** [comment]
Looks like you missed all the tests.

**Comment by dhalperi** [comment]
https://builds.apache.org/job/beam_PreCommit_MavenVerify/4178/console

**Comment by swegner** [comment]
Accidentally included extra `CTOR_DEF` rule in previous commit. Reverted.

**Comment by dhalperi** [comment]
Thanks! Merged.

**Comment by dhalperi** [comment]
(after manually inspecting travis and being sad)

**Comment by dhalperi** [comment]
@swegner Please backport to Dataflow, so that future backporting is not insane.

**Pull request closed without merging**